src/libraries/System.Collections.Immutable/tests/System.Collections.Immutable.Tests.csproj

```diff
@@ -1,14 +1,19 @@
 <Project Sdk="Microsoft.NET.Sdk">
+
   <PropertyGroup>
-    <NoWarn>$(NoWarn);0436</NoWarn>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkMinimum)</TargetFrameworks>
+    <NoWarn>$(NoWarn);0436</NoWarn>
     <DebuggerSupport Condition="'$(DebuggerSupport)' == '' and ('$(TargetOS)' == 'browser' or '$(TargetOS)' == 'wasi')">true</DebuggerSupport>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <!-- This test library intentionally references an inbox P2P as it needs the implementation, instead of the contract.
+         Suppress the NU1511 warning in the whole project as putting it on a P2P doesn't work: https://github.com/NuGet/Home/issues/14121 -->
+    <NoWarn>$(NoWarn);NU1511</NoWarn>
```

Comment:
Good issue 👍 

Another way we could handle this in runtime centrally would be to have a target that runs during restore and removes the  `PrunePackageReference`  items that overlap with `ProjectReference`.  Something like
```
  <Target Name="_RemoveProjectPrunePackageReferences" AfterTargets="AddPrunePackageReferences">
    <ItemGroup>
      <PrunePackageReference Remove="@(ProjectReference->'%(FileName)')" />
    </ItemGroup>
  </Target>
```